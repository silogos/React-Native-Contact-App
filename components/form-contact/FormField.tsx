import { Ionicons } from "@expo/vector-icons";
import { StyleSheet, TextInput, TextInputProps, View } from "react-native";

import { ThemedIcon } from "@/components/ThemedIcon";
import { ThemedView } from "@/components/ThemedView";
import { useThemeColor } from "@/hooks/useThemeColor";

interface FormFieldInterface extends TextInputProps {
  iconName?: keyof typeof Ionicons.glyphMap;
}

export default function FormField({ iconName, ...props }: FormFieldInterface) {
  const inputColor = useThemeColor({}, "input");
  const inputPlaceholder = useThemeColor({}, "inputPlaceholder");
  const borderColor = useThemeColor({}, "border");

  return (
    <ThemedView
      colorName="card"
      style={{ ...styles.container, borderColor: borderColor }}
    >
      {iconName ? (
        <ThemedIcon colorName="input" size={22} name={iconName} />
      ) : (
        <View style={styles.blankIcon} />
      )}
      <View style={styles.inputWrapper}>
        <TextInput
          style={{
            ...styles.input,
            color: inputColor,
          }}
          placeholderTextColor={inputPlaceholder}
          {...props}
        />
      </View>
    </ThemedView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    alignItems: "center",
    height: 58,
    paddingHorizontal: 12,
    borderTopWidth: 1,
    borderBottomWidth: 1,
    gap: 12,
  },
  blankIcon: {
    width: 22,
  },
  inputWrapper: {
    flex: 1,
    justifyContent: "center",
  },
  input: {
    fontSize: 18,
  },
});
