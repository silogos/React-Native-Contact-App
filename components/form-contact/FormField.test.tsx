import React from "react";
import { fireEvent, render } from "@testing-library/react-native";

import FormField from "./FormField";

describe("FormField", () => {
  it("renders correctly", () => {
    const elm = render(<FormField iconName="accessibility" />).toJSON();

    expect(elm).toMatchSnapshot();
  });

  it("change value", () => {
    const testID = "text-input";
    const onChangeText = jest.fn();
    const { getByTestId } = render(
      <FormField
        testID={testID}
        iconName="accessibility"
        onChangeText={onChangeText}
      />
    );

    const input = getByTestId(testID);

    fireEvent.changeText(input, "Cobain ubah");
    expect(onChangeText).toHaveBeenCalledWith("Cobain ubah");
  });
});
