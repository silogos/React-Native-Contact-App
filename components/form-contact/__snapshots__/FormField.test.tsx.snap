// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#e4e4e7",
        "borderTopWidth": 1,
        "flexDirection": "row",
        "gap": 12,
        "height": 58,
        "paddingHorizontal": 12,
      },
    ]
  }
>
  <Text />
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <TextInput
      placeholderTextColor="#343e4e"
      style={
        {
          "color": "#1e293b",
          "fontSize": 18,
        }
      }
    />
  </View>
</View>
`;
